apps/vaporgui/NavigationEventRouter.cpp

```diff
@@ -1004,20 +1004,38 @@ void NavigationEventRouter::_setViewpointParams(
 	const double dirvec[3], const double upvec[3]
 ) const {
 
+    std::vector<double> modelview, centerv;
+
 	// Ugh. Use trackball to convert viewing vectors into a model view
 	// matrix 
 	//
 	Trackball trackball;
-    trackball.setFromFrame(posvec, dirvec, upvec, center, true);
-    trackball.TrackballSetMatrix();
-
-	const double *m = trackball.GetModelViewMatrix();
-
-	vector <double> modelview (m, m + 16);
-	vector <double> centerv (center, center + 3);
+    int rc = trackball.setFromFrame(posvec, dirvec, upvec, center, true);
+    if (rc < 0 ) {  // If trackball fails
+
```

Comment:
Why is recovery needed here? If setFromFrame fails why not simply return and leave the params database unchanged?

Reply:
This was to refresh the gui with valid values.  I agree that this shouldn't be modifying the params database.  It now just refreshes the gui with what's in params.